src/middleware.ts

```diff
@@ -287,6 +287,7 @@ type DeepPartial<T extends Object> = {
 export type StateStorage = {
   getItem: (name: string) => string | null | Promise<string | null>
   setItem: (name: string, value: string) => void | Promise<void>
+  removeItem: (name: string) => void | Promise<void>
```

Comment:
@dai-shi FYI for the changelog: `persist` now require a storage with a `removeItem` method. That's kind of a breaking change, especially if the user gives a custom storage.

Reply:
Can we have a migration path? In v3, make it optional and, warn if it's not there. And in v4, make it required.